editor/editor_main_screen.cpp

```diff
@@ -238,13 +245,12 @@ bool EditorMainScreen::can_auto_switch_screens() const {
 		return true;
 	}
 	// Only allow auto-switching if the selected button is to the left of the Script button.
-	for (int i = 0; i < button_hb->get_child_count(); i++) {
-		Button *button = Object::cast_to<Button>(button_hb->get_child(i));
-		if (button->get_text() == "Script") {
+	for (const EditorPlugin *plugin : editor_table) {
```

Comment:
This intentionally used the button order instead of the plugin order for https://github.com/godotengine/godot/pull/104010.

I guess for that we would need a way to support reordering them.

Reply:
I went through all the relevant code before making this change. Almost all code in `EditorNode` and `EditorMainScreen` are using button index and main screen plugin index interchangeably.

Changing to comparing display names makes these names part of logic, defeating the purpose of being for display only.

Before there is a proper solution for reordering the buttons, I think assuming button index not being equal to plugin index only in this function makes the code unnecessarily complicated and confusing.